applications/portal/backend/api/views/rest/experiment.py

```diff
@@ -217,3 +224,73 @@ def download_populations(self, request, pk=None, active_populations=None):
 
     def perform_create(self, serializer):
         experiment = serializer.save(owner=self.request.user)
+
+    def check_if_user_is_owner(self, kwargs_experiment_id):
+        try:
+            user = get_current_user()
+            experiment = Experiment.objects.get(id=kwargs_experiment_id)
+            if user.id != experiment.owner.id:
+                return Response(status=status.HTTP_403_FORBIDDEN, data={'detail': 'You are not the owner of the experiment'})
+        except Experiment.DoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND, data={'detail': 'Experiment does not exist'})
+        
+    def check_if_subpopulation_exists(self, subpopulation_id, kwargs_experiment_id):
+        try:
+            sub_population_object = Population.objects.get(id=subpopulation_id)
+            is_residential = sub_population_object.experiment is None
+            if is_residential:
+                return Response(status=status.HTTP_403_FORBIDDEN, data={'detail': 'You cannot change the name of a residential population'})
+            if int(sub_population_object.experiment.id) != int(kwargs_experiment_id):
+                return Response(status=status.HTTP_404_NOT_FOUND, data={'detail': f'Subpopulation with id {subpopulation_id} does not exist in this experiment'})
+        except Population.DoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND, data={'detail': f'Subpopulation with id {subpopulation_id} does not exist'})
+        return sub_population_object
+    
+
+    @action(
+        detail=True, 
+        methods=['patch'], 
+        parser_classes=(JSONParser,),
+        url_name="rename_population"
+    )
+    def rename_population(self, request, **kwargs):
+        kwargs_experiment_id = kwargs['pk']
+        is_user_owner = self.check_if_user_is_owner(kwargs_experiment_id)
+        request_data = json.loads(request.body)
```

Comment:
@D-GopalKrishna here you are mixing serializers and json bodies.... please pick either use serializers or use the raw body